great_expectations/datasource/new_datasource.py

```diff
@@ -204,6 +208,35 @@ def get_batch_list_from_batch_request(
                 batches.append(new_batch)
             return batches
 
+    def _update_missing_data_asset_name(
+        self, batch_request: Union[BatchRequest, RuntimeBatchRequest]
+    ) -> None:
+        # if the data_asset_name is missing from a configured data connector config, add it as a table_name
```

Comment:
@NathanFarmer @talagluck Could you please explain why this is a proper thing to do?  Is it possible to report on this `data_asset` (which is rightfully discovered during the inference step) without storing it in the configuration's memory?  Thanks.

Reply:
@alexsherstinsky I think you already understand, judging by your comment, that this step takes the batch_request passed at runtime and adds it to the in-memory data context config. Once this step is complete, all we need to do is call `data_context._save_project_config` to take that in-memory config and save it to `great_expectations.yml`. This is an important requirement. What I am hoping to discover is a more "proper" way to do this, but as of now the codepath is not clear to me.

Reply:
@NathanFarmer I am quite uneasy about the "side effect" like nature of this process, but understand the rationale.  It would be nice if assets did not just get saved automatically -- it would be nice if such operations were clear to the user and under the user's control.

Reply:
@alexsherstinsky I was able to refactor the data connector conditional to make it more proper.

This behavior does not occur for `InferredAssetSqlDataConnector`s. The behavior is meant to signal to the user how data connectors are meant to be configured.